### DISCUSSION TITLE
curl multi Asynchronous Usage Questions Asked

### DISCUSSION THREAD

**Discussion by ghost** [curl/curl#12656] [Q&A] [answered]
When I remove an easy_handle using curl_multi_remove_handle curl_multi_info_read will not be able to read the associated removal message.

My requirement is whether I can get the message that easy_handle is removed when I remove easy_handle.

In other words, when an easy_handle is executing, he is establishing a connection with the other tcp server. curl can set the connection timeout to any value, for example, I set the curl connection timeout to 10s, then I want to stop the easy_handle early.

Of course curl_easy_setopt(base->curl_easy_handle, CURLOPT_PROGRESSFUNCTION) is not triggered in real time when it encounters a server that is not linkable, so I can't do the cancel method here.

**Comment by jay** [comment, marked as answer]
> My requirement is whether I can get the message that easy_handle is removed when I remove easy_handle.
> 
> In other words, when an easy_handle is executing, he is establishing a connection with the other tcp server. curl can set the connection timeout to any value, for example, I set the curl connection timeout to 10s, then I want to stop the easy_handle early.

What message are you looking for? Yes, it's acceptable to stop an in-use easy handle by removing it from a multi.
